mitiq/lre/tests/test_multivariate_richardson.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Unit tests for multivariate extrapolation inference functions."""
+
+from math import comb
+
+import numpy as np
+import pytest
+from cirq import Circuit, LineQubit, ops
+
+from mitiq.lre.inference.multivariate_richardson import (
+    _create_variable_combinations,
+    _get_variables,
+    full_monomial_basis,
+    linear_combination_coefficients,
+    sample_matrix,
+)
+from mitiq.lre.multivariate_scaling.layerwise_folding import (
+    _get_num_layers_without_measurements,
+    _get_scale_factor_vectors,
+)
+
+
+@pytest.mark.parametrize(
+    "test_num_layers, expected_list",
+    [
+        (2, ["λ_1", "λ_2"]),
+        (3, ["λ_1", "λ_2", "λ_3"]),
+        (4, ["λ_1", "λ_2", "λ_3", "λ_4"]),
+    ],
+)
+def test_get_variables(test_num_layers, expected_list):
+    calculated_variables = _get_variables(test_num_layers)
+    assert len(calculated_variables) == test_num_layers
+    assert calculated_variables == expected_list
+
+
+@pytest.mark.parametrize(
+    "test_num_layers, test_degree",
+    [(2, 2), (3, 2), (4, 2), (2, 3), (3, 3)],
```

Comment:
If the test timing is not too unreasonable, I think we should include tests for higher layers and degrees.